src/main/java/org/example/commands/AddUserCommand.java

```diff
@@ -0,0 +1,26 @@
+package org.example.commands;
+
+import java.util.Set;
+import org.example.exceptions.InvalidArgument;
+import org.example.model.User;
+import org.example.services.AddUserDetailsServiceInterface;
+
+public class AddUserCommand implements UserCommandInterface {
+  private final AddUserDetailsServiceInterface addUserDetails;
+
+  public AddUserCommand(AddUserDetailsServiceInterface addUserDetails) {
+    this.addUserDetails = addUserDetails;
+  }
+
+  @Override
+  public void execute(Set<User> users) {
+    User newUser = addUserDetails.newUser();
+    if (users.stream().noneMatch(user -> user.getRollNumber().equals(newUser.getRollNumber()))) {
```

Comment:
can we move this rollnumber logic to AddUserDetailsServiceInterface? because its core logic of addUserDetails